### PR TITLE
[PM-8144] Migrate auto-fill policy logic from Tab to new Vault component

### PR DESCRIPTION
**Pull request by gbubemismith**
## 🎟️ Tracking
[PM-8144](https://bitwarden.atlassian.net/browse/PM-8144)

## 📔 Objective
The current Auto-fill policy relies on the user opening the Tab page to show the toast communicating that auto-fill on page load was turned on. There is a bug with the [initially implemented behavior](https://bitwarden.atlassian.net/browse/AC-1046) that overrides any user preference anytime the user visits the Tab page–the AC team decided to keep that behavior. With this in mind, we should update the implementation as part of the Extension refresh to be a bit more seamless

## 📸 Screenshots

https://github.com/user-attachments/assets/34bc967e-c0e0-4ba2-9665-cc968a9aff8a

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-8144]: https://bitwarden.atlassian.net/browse/PM-8144?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/common/src/autofill/services/autofill-settings.service.ts [hunk 1]
```diff
@@ -146,9 +146,8 @@ export class AutofillSettingsService implements AutofillSettingsServiceAbstracti
     this.autofillOnPageLoadPolicyToastHasDisplayedState = this.stateProvider.getActive(
       AUTOFILL_ON_PAGE_LOAD_POLICY_TOAST_HAS_DISPLAYED,
     );
-    this.autofillOnPageLoadPolicyToastHasDisplayed$ = this.autofillOnPageLoadState.state$.pipe(
```
Comment (gbubemismith) [inline libs/common/src/autofill/services/autofill-settings.service.ts:149]: I discovered what looks like a bug to me: autofillOnPageLoadPolicyToastHasDisplayed$ was looking at the autofillOnPageLoadState

File: apps/browser/src/autofill/services/autofill.service.ts [hunk 2]
```diff
@@ -570,6 +570,19 @@ export default class AutofillService implements AutofillServiceInterface {
     return totpCode;
   }
 
+  /**
+   * Activates the autofill on page load org policy.
+   */
+  async setAutoFillOnPageLoadOrgPolicy(): Promise<void> {
```
Comment (gbubemismith) [inline apps/browser/src/autofill/services/autofill.service.ts:576]: Is it okay to have this in the autofill service?

Comment (cagonzalezcs) [reply apps/browser/src/autofill/services/autofill.service.ts:576]: Hmm... I don't think there is anything inherently wrong with doing that... but more than likely, it is unnecessary. 

The `AutofillSettingsService` is passed as a dependency within `services.modules.ts`... you could just instantiate that services as a dependency within the constructor for the component that needs this. 

I'm open to what @jprusik might suggest on this. He has been tackling the majority of the vault refactor work.

**Review by cagonzalezcs — DISMISSED** [review]

File: apps/browser/src/autofill/services/autofill.service.ts (hunk 2 above)
Comment (jprusik) [reply apps/browser/src/autofill/services/autofill.service.ts:576]: I think it is in line with our current strategy for settings state (and I do think our team should own it), but I could see a unified policy service in the future. Until then, this works for me 👍

**Review by jprusik — APPROVED** [review]

**Review by jprusik — APPROVED** [review]

**Pull request merged**